### ISSUE TITLE
No CPPM signal with FrSky D8R-II Plus

### ISSUE THREAD

**Issue by dronehitech** [MegaPirateNG/ardupilot-mpng#154]
Hello,

I dont get my HK MultiWii Megapirate AIO working with my FrSky D8R-II Plus with CPPM signal, if I use the regular way to flash my board through arduino. My "RCInput_MPNG.cpp" seems to be ok in my opinion, but it doesnt work.

If I use the FlashTool instead, and chose the CPPM there, then it works. Any ideas?
Thanks!

**Comment by SirAlex** [comment]
please, show us your file RCInput_MPNG.cpp. Upload it somewhere or send it to me rsoft@tut.by

**Comment by dronehitech** [comment]
Here is a link to the file: https://www.dropbox.com/s/htyyrumu9vomlga/RCInput_MPNG.cpp

**Comment by smurfy** [comment]
Quite strange, your RCInput_MPNG.cpp looks fine. 
The Flashtool uses the same code (just a different gcc compiler) 
Even if you don't set the PPM option there it should compile with PPM because that is the default.

Try cloning the r4 branch again and start fresh. Other than that i'm out of ideas.

**Comment by dronehitech** [comment]
Yeah, it is strange :) I even downloaded it again and flashed the newly downloaded version, but without success. Erased EEPROM, reboot and so on - without success.

**Comment by sreichholf** [comment]
Have you correctly flashed your D8R-II Plus with the fixed CPPM Firmware for the XP?
I am using the extact same receiver (with XP Firmware) with CPPM without any issues here.

**Comment by dronehitech** [comment]
Yes. As said it works fine when uploading MegaPirate with the FlashTool and not when uploading through Arduino Sketch.
